app-drawer/app-drawer.html

```diff
@@ -260,75 +260,198 @@
+
           switch (event.detail.state) {
             case 'start':
-              // Disable transitions since styles will reflect user track events.
-              this.$.contentContainer.style.transitionDuration = '0s';
-              this.$.scrim.style.transitionDuration = '0s';
-              this.$.scrim.style.visibility = 'visible';
-
-              if (this.opened) {
-                this._translateOffset = 0;
-              } else if (this.position == 'left') {
-                this._translateOffset = -this.getWidth();
-              } else if (this.position == 'right') {
-                this._translateOffset = this.getWidth();
-              }
+              this._trackStart(event);
               break;
-
             case 'track':
-              var translateDelta = event.detail.dx + this._translateOffset;
-              switch (this.position) {
-                case 'left':
-                  translateDelta = Math.min(0, translateDelta);
-                  this._translateDrawer(translateDelta);
-                  this.$.scrim.style.opacity = 1 + translateDelta / this.getWidth();
-                  break;
-                case 'right':
-                  translateDelta = Math.max(0, translateDelta);
-                  this._translateDrawer(translateDelta);
-                  this.$.scrim.style.opacity = 1 - translateDelta / this.getWidth();
-                  break;
-              }
-
-              // Keep track of the last user track event with non-zero ddx (which we
-              // need to determine whether to keep the drawer open/closed) since the
-              // very last track event may have zero ddx.
-              if (event.detail.ddx !== 0) {
-                this._lastTrackDirection = event.detail.ddx;
-              }
+              this._trackMove(event);
               break;
-
             case 'end':
-              this.$.contentContainer.style.transitionDuration = '';
-              this.$.scrim.style.transitionDuration = '';
-              this.$.scrim.style.visibility = '';
-              this.$.scrim.style.opacity = '';
-              this.transform('', this.$.contentContainer);
-
-              // Only toggle the drawer if the user intentionally swipes in one direction.
-              if (Math.abs(event.detail.dx) > 30) {
-                this.opened = (this.position === 'left' && this._lastTrackDirection > 0) ||
-                  (this.position === 'right' && this._lastTrackDirection < 0);
-              }
+              this._trackEnd(event);
               break;
           }
         }
       },
 
+      _trackStart: function(event) {
+        // Disable transitions since style attributes will reflect user track events.
+        this.$.contentContainer.style.transitionDuration = '0s';
+        this.$.scrim.style.transitionDuration = '0s';
+        this.$.scrim.style.visibility = 'visible';
+
+        var rect = this.$.contentContainer.getBoundingClientRect();
+        if (this.position == 'left') {
```

Comment:
minor:  `this._translateOffset = this.position == 'left' ? rect.left : rect.right - window.innerWidth;`